src/xenia/kernel/kernel_state.cc

```diff
@@ -138,7 +138,17 @@ void KernelState::set_process_type(uint32_t value) {
 
 uint32_t KernelState::AllocateTLS() { return uint32_t(tls_bitmap_.Acquire()); }
 
-void KernelState::FreeTLS(uint32_t slot) { tls_bitmap_.Release(slot); }
+void KernelState::FreeTLS(uint32_t slot) {
+  const std::vector<object_ref<XThread>> threads =
+      kernel_state()->object_table()->GetObjectsByType<kernel::XThread>();
```

Comment:
You shouldn't need to have to do `kernel_state()` here, you're already in `KernelState`.

Reply:
Fair point. That's habit for me at this point 😅 Fixed